lib/src/unresolved_type.dart

```diff
@@ -10,7 +10,11 @@ class UnresolvedType {
       3 => <A, B, C>(f) => f<UnresolvedType>(),
       4 => <A, B, C, D>(f) => f<UnresolvedType>(),
       5 => <A, B, C, D, E>(f) => f<UnresolvedType>(),
-      _ => throw Exception('TypePlus only supports generic classes with up to 5 type arguments.'),
+      6 => <A, B, C, D, E, F>(f) => f<UnresolvedType>(),
+      7 => <A, B, C, D, E, F, G>(f) => f<UnresolvedType>(),
+      8 => <A, B, C, D, E, F, G, H>(f) => f<UnresolvedType>(),
+      9 => <A, B, C, D, E, F, G, H, I>(f) => f<UnresolvedType>(),
+      _ => throw Exception('TypePlus only supports generic classes with up to 10 type arguments.'),
```

Comment:
Nitpick, but "up to 10" would include 10. So either implement also the 10 arguments case or change to 9. Same for TypeSwitcher